### ISSUE TITLE
code review

### ISSUE THREAD

**Issue by YasserRabee** [ALITO1998/todolist#1]
https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L19-L25

* We should handle known category values only. If an unknown value is set, we should raise an error.
* ele.status is very complicated for a todo list app. A todo is either pending or done, so maybe a boolean field `done` is just enough and clearer.
* can use refactor this code to use `data.filter()` only once? If you ask why, imagine we have multiple filters: category, assignee, due_date, creation_date, text search, ...etc. The first step to be able to implement all those filters (cleanly) is to use `data.filter()` only once.

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L33-L36

Why setting shownData manually here? Setting data only should be enough and the previous useEffect will handle the change.

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L40

Better to have a separate function to create a new task record. Also the caller to the addTaskHandler should be the one who creates a ready task record. This handler should be only responsible for pushing it into data.

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L43

Bad naming.

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L58-L59

Code smells:
* Component name `Content` is very generic and doesn't indicate what exactly it is.
* If content needs props to read category value, and to change it - while App component needs only to read category => category useState should be moved inside Content. 

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L67-L70

If Content shows the modal, they why defining the modal in App? App component itself doesn't need to know about the model. Maybe it is the `addTaskHandler` that confused you? Pass it as a prop to Content - the same way you did with delete, checked, ....

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/AddTaskModal/AddTask/AddTask.js#L5

This component shouldn't know it is part of a modal. You may need to rename `closeModal` to `onCancel` and `AddTaskHandler` to `onSave`. If there is no cancel button, so onSave should be enough, and the modal wrapper should call closeModal after save.

Also you need to move this component to another directory.

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/AddTaskModal/AddTask/AddTask.js#L28-L30

Why using a FormControl for an error message? Please see how MUI handle formControl validations and error messages https://mui.com/material-ui/react-text-field/#validation

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/Content/TaskList/TaskList.js#L7

Why passing prop named `id`?

https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/Content/Task/Task.js#L18-L33

what?

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L19-L25
>```
if (category === "all") {
            setShownData(data);
        } else {
            setShownData(data.filter((ele) => ele.status === category));
        }
```

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L33-L36
>Had been corrected in next commit

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L40
> Need more explain

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L43
>Agree with you , I will change it

**Comment by ALITO1998** [comment]
> https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L58-L59
>Need more explain for correct naming
>I try to collect all "useState" and "Handlers" in app file

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/App/App.js#L67-L70
>I try to separate "modal" from "content".
>and keep every  "Handlers" in app file.

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/AddTaskModal/AddTask/AddTask.js#L5
>Had been corrected in next commit
>I named props the same name of their content to make it easy to track it .
>Need more explain why need to move this component to another directory!!

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/AddTaskModal/AddTask/AddTask.js#L28-L30
>Thank you for this guide.

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/Content/TaskList/TaskList.js#L7
>Thanks ,I delete it.

**Comment by ALITO1998** [comment]
>https://github.com/ALITO1998/todolist/blob/bb7a57f5b26cf62cfe432aa63a4d4e00e4237ef8/src/Components/Content/Task/Task.js#L18-L33
>Had been corrected in next commit

**Comment by ALITO1998** [comment]
>>Thank you again for your time
>>جزاكم الله خيرا

**Comment by ALITO1998** [comment]
Every issues had been solved in commit "issue 1 (solved)"
